core/corehttp/gateway.go

```diff
@@ -212,6 +200,28 @@ var defaultKnownGateways = map[string]*gateway.Specification{
 	"localhost": subdomainGatewaySpec,
 }
 
+func getGatewayConfig(n *core.IpfsNode) (gateway.Config, error) {
+	cfg, err := n.Repo.Config()
+	if err != nil {
+		return gateway.Config{}, err
+	}
+
+	headers := make(map[string][]string, len(cfg.Gateway.HTTPHeaders))
+	for h, v := range cfg.Gateway.HTTPHeaders {
+		headers[http.CanonicalHeaderKey(h)] = v
+	}
+	gateway.AddAccessControlHeaders(headers)
+
+	gwConfig := gateway.Config{
+		Headers:        headers,
+		TrustedMode:    !cfg.Gateway.OnlyTrustless.WithDefault(config.DefaultOnlyTrustless),
```

Comment:
Why is this `!`ed ? Even if this is correct the name of one of the two fields is weird then.

Reply:
@Jorropo this is correct. Boxo is trustless by default, so you set `TrustedMode`. Kubo is trusted by default, so you set `OnlyTrustless`. Trusted = !Trustless

Reply:
I see, it's weird to have to names and behaviour that are opposites in Kubo and Boxo.
It's not a deal breaker, but I would consider changing one of them.

Reply:
The option I see to make it uniform is changing Kubo's `OnlyTrustless` to `TrustedMode` and defaulting that to `true` in order to not change the current behaviour. I definitely agree that that could make the code clear. I will come back tot his today and likely make that change.

Reply:
I changed it to match Boxo's option.